scenario/consistency_checker.py

```diff
@@ -348,11 +359,16 @@ def check_config_consistency(
         converters = {
             "string": str,
             "int": int,
-            "integer": int,  # fixme: which one is it?
-            "number": float,
+            "float": float,
             "boolean": bool,
             # "attrs": NotImplemented,  # fixme: wot?
         }
+        if juju_version >= (3, 4):
```

Comment:
Maybe this should be 3.1? I didn't do any investigation myself (other than verifying things on 3.4.0) but [John did in this thread](https://github.com/canonical/istio-operators/issues/380#issuecomment-2020861148) and it seems like 3.1 will accept the `type:secret` even though it's not really _useful_ until `3.4`.

Reply:
Yeah, this won't be useful till user secrets (in 3.4?), so probably 3.4 is better here.